File: src/baekjoon/수학/수찬은_마린보이야_15921/main.py
'''
5
5 10 10 15 20 # 5*(1/5) + 10*(2/5) + 15*(1/5) + 20*(1/5)
'''

n = int(input())
if n:
    l = list(map(int, input().split()))
    r = sorted(set(l)) # 중복값을 제거한 리스트
    avg = sum(l)/len(l) # 평균
    exp = 0
    #
    for i in r:
        exp += i * (l.count(i)/n)

    print('%.2f' % (avg / exp))
else:
    print("divide by zero")

File: src/baekjoon/수학/수찬은_마린보이야_15921/Main.java
package baekjoon.수학.수찬은_마린보이야_15921;

/**
5
5 10 10 15 20
 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Map<Integer,Integer> map = new HashMap<>();
        if(n == 0){
            System.out.println("divide by zero");
        }else{
            StringTokenizer st = new StringTokenizer(br.readLine());
            int[] arr = new int[n];
            // 합계 구하기
            int sum = 0;
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
                sum += arr[i];
                map.put(arr[i], map.getOrDefault(arr[i],0) + 1);
            }
            // 평균 값
            double avg = sum / (double)n;

            // 기대값
            double exp = 0.00;
            for (int item:map.keySet()) {
                exp += item * map.get(item) / (double)n;
            }
            System.out.printf("%.2f", avg/exp);


        }
    }

    public static void main(String[] args) throws IOException{
        Main m = new Main();
        m.solution();
    }

}